src/Playwright/Transport/Converters/EvaluateArgumentValueConverter.cs

```diff
@@ -249,7 +249,15 @@ private static object ParseEvaluateResult(JsonElement result, Type t, JsonSerial
                     var defaultConverter = JsonExtensions.GetNewDefaultSerializerOptions();
                     string serialized = JsonSerializer.Serialize(dicResult, defaultConverter);
 
-                    return JsonSerializer.Deserialize<T>(serialized, defaultConverter);
+                    var retValue = JsonSerializer.Deserialize<T>(serialized, defaultConverter);
+
+                    // This particular check & clone call is required because otherwise, when we attempt to
+                    // access the value, we run into an Object Disposed Exception.
+                    // See: https://github.com/microsoft/playwright-dotnet/issues/1706
+                    if (retValue is JsonElement retValueAsJson)
+                    {
+                        return retValueAsJson.Clone();
```

Comment:
Is this actually necessary? I'd expect original `retValue`  to have lifteme independent of any documents.